src/sinks/gcp/stackdriver_metrics.rs

```diff
@@ -92,162 +98,346 @@ fn default_endpoint() -> String {
+        let mut time_tracker = StartTimeTracker::new();
+        let mut last_time = Instant::now();
+        let mut buffer = Vec::<MetricDataPoint>::new();
+        let mut client =
+            MetricServiceClient::<Channel>::with_interceptor(channel, interceptor(auth));
+
+        loop {
+            if aggregate.len() == GCP_MAX_BATCH_SIZE {
+                // In case we already accumulated the maximum unique data point count supported by
+                // GCP, we wait so we are not denied by the GCP backend for sending too many
+                // requests.
+                while last_time.elapsed() < GCP_SMALLEST_CREATE_TIMESERIES_CALL_FREQUENCY {
+                    tokio::time::sleep(Duration::from_millis(500)).await;
+                }
+            } else {
+                while let Some(data_point) = buffer.pop() {
+                    aggregate
+                        .entry(format!("{}/{}", data_point.namespace, data_point.name))
+                        .and_modify(|cur| cur.merge(&data_point))
+                        .or_insert(data_point);
+
+                    if aggregate.len() == GCP_MAX_BATCH_SIZE {
+                        break;
+                    }
+                }
+            }
+
+            // We send a timeseries create request as long as we have at least one metric data point and
+            // we waited enough since the last time we sent something to GCP.
+            if !aggregate.is_empty()
+                && last_time.elapsed() >= GCP_SMALLEST_CREATE_TIMESERIES_CALL_FREQUENCY
+            {
+                let start_time = time_tracker.start_time();
+                let time_series = aggregate
+                    .drain()
+                    .map(|t| gcp_timeseries(&self.config, start_time, t.1))
+                    .collect::<Vec<_>>();
+
+                let req = googleapis_proto::CreateTimeSeriesRequest {
+                    name: format!("projects/{}", self.config.project_id),
+                    time_series,
                 };
 
-                (*value, interval, gcp::GcpMetricKind::Cumulative)
+                let mut attempts = 1usize;
+
+                loop {
+                    if let Err(status) = client.create_time_series(Request::new(req.clone())).await
+                    {
+                        if status.code() == Code::Internal
+                            || status.code() == Code::Unknown
+                                && attempts < MAX_RETRIES_ON_LEGITIMATE_GCP_ERROR
+                        {
+                            tokio::time::sleep(Duration::from_millis(500)).await;
+                            attempts += 1;
+                            continue;
+                        }
+
+                        error!(message = "Error when sending time_series", %status);
+                        emit!(ComponentEventsDropped::<UNINTENTIONAL> {
+                            count: attempts,
```

Comment:
`count` here should be the number of events dropped, not the number of attempts made.